Plugins/Public/event/Main.cpp

```diff
@@ -64,7 +68,10 @@ struct TRADE_EVENT {
 	uint uHashID;
 	string sEventName;
 	string sURL;
+	wstring eventDescription;
 	int iBonusCash;
+	bool isClassRestricted = false;
+	wstring allowedShipClass;
```

Comment:
are we not likely to want to allow multiple ship classes?

Reply:
Definitely. Personally I'd suggest making either a vector or unordered_set of allowed transport classes. If empty, allow event for all ship types. Otherwise check against the set and proceed only on match.

Reply:
No we don't want multiple ship classes, otherwise it wouldn'be tailored profit-wise for that class

Reply:
If a route is tailored to trains, there's no reason that I can see to stop smaller transports or liners from participating despite having lower CSU.

Reply:
> If a route is tailored to trains, there's no reason that I can see to stop smaller transports or liners from participating despite having lower CSU.

In that scenario, you don't even need to make a shipclass restricted run. Cuz running anything smaller will be pointless

Reply:
Thesse really have two usages:
Cute RP focused thing (Freighter smuggling xyz out of a warzone for instance)
Giving actual profitability to specific underplayed ship classes. i.e. Liners, Freighters